### PR TITLE
Fix number of rows in randomly generated lists columns

### PR DESCRIPTION
**Pull request by vuule**
## Description
Changing `single_level_mean` to double introduced a rounding error in the iterative process of generating random lists columns. This PR addressed the issue by enforcing the correct row count in the root lists column.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Review by shrshi — APPROVED** [review]
Thank you for the bug fix, looks good to me! :)

File: cpp/benchmarks/common/generate_input.cu [hunk 1]
```diff
@@ -751,13 +752,16 @@ std::unique_ptr<cudf::column> create_random_column<cudf::list_view>(data_profile
 
   // Generate the list column bottom-up
   auto list_column = std::move(leaf_column);
-  for (int lvl = 0; lvl < dist_params.max_depth; ++lvl) {
+  for (int lvl = dist_params.max_depth; lvl > 0; --lvl) {
```
Comment (ttnghia) [inline cpp/benchmarks/common/generate_input.cu:755]: Why are we moving backward?

Comment (vuule) [reply cpp/benchmarks/common/generate_input.cu:755]: We've always generated these columns bottom-up, just the value of `lvl` did not match this. `lvl` is not used in the loop, outside of the new check.

Comment (ttnghia) [reply cpp/benchmarks/common/generate_input.cu:755]: Ah that makes sense.

**Review by ttnghia — APPROVED** [review]

File: cpp/benchmarks/common/generate_input.cu [hunk 2]
```diff
@@ -740,7 +740,8 @@ std::unique_ptr<cudf::column> create_random_column<cudf::list_view>(data_profile
 {
   auto const dist_params       = profile.get_distribution_params<cudf::list_view>();
   auto const single_level_mean = get_distribution_mean(dist_params.length_params);
-  auto const num_elements      = num_rows * pow(single_level_mean, dist_params.max_depth);
+  cudf::size_type const num_elements =
+    std::lround(num_rows * pow(single_level_mean, dist_params.max_depth));
```
Comment (bdice) [inline cpp/benchmarks/common/generate_input.cu:744]:
```suggestion
    std::lround(num_rows * std::pow(single_level_mean, dist_params.max_depth));
```

**Review by bdice — APPROVED** [review]

**Pull request merged**